Tensile/Utilities/merge.py

```diff
@@ -73,8 +95,8 @@ def removeUnusedKernels(origData):
             if index == oldSolIndex:
                 origData[7][item][1][0] = i
 
-    print("Removed ",len(unusedKernels), " unused kernels from output logic file")
-    return origData
+    numInvalidRemoved = origNumSolutions - len(origData[5])
```

Comment:
Use a solutionOldToNewIdxMap { oldIdx : newIdx }, init map [oldIdx] = -1 as not referenced
Loop origData[7] (problems) and mark the referenced solution to 1 ( map[oldIdx] = 1) as referenced
Loop origData[5] (solutions), skip if [oldIdx] = -1. Set the valid one a newIdx, update the map[oldIdx] = newIdx
Loop origData[7] (problems) re-index the oldIdx to newIdx
Maybe this can avoid some nested loops with larger O(M*N) to (M+N)